### PR TITLE
mac: Setting to explicitly choose pan or zoom for mouse wheel (#1339)

### PR DESCRIPTION
**Pull request by nanoant**
While auto remains the default, we now support explicit choice of doing pan or zoom when scrolling with mouse wheel or touch pad.

This can be also adapted for other (non-macOS) platforms.

### CODE REVIEW

**Comment by vespakoen** [comment]
Wouldn't it make more sense to use something like:
```
        AUTO,
        MOUSE,
        TRACKPAD,
```
?

Or maybe I am mistaken and this is not related to https://github.com/solvespace/solvespace/issues/1339 but an extra feature?

**Comment by phkahler** [comment]
I'm with @vespakoen on this. It also should not be platform specific. And pan with scroll wheel doesn't make sense because it only works vertically. Scroll means zoom.

**Comment by nanoant** [comment]
@phkahler @vespakoen I can reiterate on this, what should be then the correct name for the setting?
`Pointing device (*) Auto-detect ( ) Mouse ( ) Trackpad` ?

**Comment by phkahler** [comment]
I should have been clearer, but sometimes I don't want to upset contributors. I don't want scroll wheel to pan even as an option. Won't merge it.

**Comment by nanoant** [comment]
@phkahler No problem for me. And yes this is a follow up for #1339 and #1093 e1b0784b31c157922be8a618a4247b26a9fd642d which started all of this by introducing panning behavior on trackpads implemented inside of `- (void)scrollWheel:(NSEvent *)nsEvent`.

Please note that the event is called "scroll wheel", not "trouch" or "track pad". So its purpose is to handle scrolling not anything else. This implementation broke my Apple Magic Mouse workflow that suddenly started panning instead of zooming. I tried to un-break it by extending too weak `nsEvent.subtype == NSEventSubtypeTabletPoint && kind == Platform::Window::Kind::TOPLEVEL` check, adding `activeTrackpadTouches > 0`. But apparently it stills trips on some weird Logitech mice setups. I tested latest f16c692803396291c12b4c0d9ed8c8bcc7af0ac7 with my two Logitech mice and some 2 Chinese unbranded one, and they all worked fine. But I don't use any Logitech customization software and I can only suspect such a software may make USB HID look like touch pad to `guimac.m` code for whatever reason.

So long story short, I am trying to make this software work for all macOS users, but I am not responsible for having #1339 guys complaining that wheel is panning instead of moving, because I did not contribute that in the first place. (Un?)fortunately I was called up with my GitHub id, because I was last person trying to fix the flawed #1093 implementation.

If #1093 was implemented using gestures maybe we had no such a problem, but the original author decided to do it within `scrollWheel`.

So I hope this is clear. And I am not upset (contributor), but I want to make sure you get the whole picture and as an ex-macOS developer who's now doing something completely else for living and doing 3D printing for fun I am trying to help 🥲

**Comment by phkahler** [comment]
IMHO after trying a couple versions of solvespace on my old MBA, a setting isn't appropriate. We need to use modifier keys to differentiate the mouse button behaviors when using the trackpad. Shift, Control, Super, are all there. I'd even like if one of them were used so click-drag isn't needed since it's wierd on a trackpad. But this is just one opinion ATM.
